### ISSUE TITLE
Add support to enable lsp completion inside the OCaml blocks for OCamllex files

### ISSUE THREAD

**Issue by smorimoto** [ocamllabs/vscode-ocaml-platform#544]

**Comment by smorimoto** [comment]
@rgrinberg @voodoos Where should this be addressed? Should we support this on the Merlin side?

**Comment by smorimoto** [comment]
Well, I'm sure this shouldn't be addressed on the extension side.

**Comment by mnxn** [comment]
> Where should this be addressed? Should we support this on the Merlin side?

I think both the language server and Merlin would have to add support. If I'm not mistaken, LSP support would have to be registered for the entire `mll` filetype because embedded blocks are not supported by the protocol. Merlin would have to recognize regex variables defined in the lexer rules: https://caml.inria.fr/pub/docs/manual-ocaml/lexyacc.html#ss:ocamllex-variables

**Comment by smorimoto** [comment]
Yes, and it would be great if something like ocamlformat supports that, as well. I don't know if it should be done through LSP or by ocamlformat itself, but I think we need to take that into account.
